respokeSDK/src/main/java/com/digium/respokesdk/RespokeClient.java

```diff
@@ -191,6 +190,29 @@
 
 
     /**
+     * A listener interface to receive a notification when the request to retrieve the history
+     * of messages for a list of groups has completed
+     */
+    public interface GroupHistoriesCompletionListener {
+
+        void onSuccess(Map groupsToMessages);
```

Comment:
Good catch. It's better to be specific in what the key/values are. Otherwise clients have to figure it out and cast to the right types.